src/api/c/convolve.cpp

```diff
@@ -74,6 +74,8 @@ af_err convolve(af_array *out, const af_array signal, const af_array filter)
 
         AF_BATCH_KIND convBT = identifyBatchKind<baseDim>(sdims, fdims);
 
+        ARG_ASSERT(1, (sdims.ndims()>=1));
+        ARG_ASSERT(2, (fdims.ndims()>=1));
```

Comment:
If we do just that condition, how can we identify right batch operation from the [list of 4](http://arrayfire.org/docs/group__signal__func__convolve3.htm) types of batches we handle ?